src/courtroom.cpp

```diff
@@ -5243,12 +5244,16 @@ void Courtroom::on_showname_enable_clicked()
 void Courtroom::regenerate_ic_chatlog()
 {
   ui_ic_chatlog->clear();
+  last_ic_message = "";
   foreach (chatlogpiece item, ic_chatlog_history) {
-    append_ic_text(item.get_message(),
-                   ui_showname_enable->isChecked() ? item.get_showname()
-                                                   : item.get_name(),
+    QString message = item.get_message();
```

Comment:
(We discussed that this is trimmed beforehand, so all blankposts are guaranteed to be empty strings.)